compiler/src/dotty/tools/dotc/util/CommentParsing.scala

```diff
@@ -172,8 +172,10 @@ object CommentParsing {
   def skipVariable(str: String, start: Int): Int = {
     var idx = start
     if (idx < str.length && (str charAt idx) == '{') {
-      do idx += 1
-      while (idx < str.length && (str charAt idx) != '}')
+      while {
+        idx += 1
+        idx < str.length && (str charAt idx) != '}'
+      } do ()
```

Comment:
For this PR we should only remove `do .. while` but not add `while .. do`

Reply:
It's actually already supported by the existing code. I converted to a mixture of the two to test both forms.